022-feature-gates.md

```diff
@@ -0,0 +1,125 @@
+# Feature Gates
+
+This proposal suggests to add support for Feature Gates in the Strimzi operators.
+Feature Gates should give us additional options how to control and mature different behaviors in the operators.
+
+## Current situation
+
+Currently, any features or behaviors of the operator are controlled by one of these two mechanisms:
+
+* The Strimzi custom resources (as a form of an API)
+* The state of the Kubernetes resources
+
+The following examples show how are these mechanisms used to give a better idea:
+
+* The `strimzi.io/use-connector-resources` annotation in the `KafkaConnect` custom resource defines whether the Connector Operator should be enabled or not for given Kafka Connect resource.
+* The `useServiceDnsDomain` in the listener configuration of the `Kafka` custom resource defines what kind of address should be used for given listener.
+  Whether fully qualified DNS name or not.
+* When the operator during Kafka upgrade sees that the old pods are still using the TLS sidecars, it will automatically loosen the TLS configuration to make sure the upgrade can be completed.
+
+## Motivation
+
+In many cases, the mechanisms described above work fine and are well suited.
+But there are also limitations:
+
+* Changes to the custom resources APIs have permanent character.
```

Comment:
```suggestion
* Changes to the custom resources APIs have upgrade and forwards compatibility concerns.
```
Just reworded as I understand what it was getting at, but also you then refute its permanence in the next line, I think being more specific here is better, but your point is still well delivered.